servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HttpRequestDecoder.java

```diff
@@ -113,11 +104,11 @@ private static HttpRequestMethod decodeHttpMethod(final ByteBuf buffer, final in
             return method;
         }
         try {
-            buffer.forEachByte(start, length, ENSURE_UPPER_CASE);
-        } catch (IllegalCharacterException cause) {
-            throw newInvalidMethodException(cause);
+            return HttpRequestMethod.of(methodName, NONE);
+        } catch (IllegalArgumentException cause) {
+            throw new StacklessDecoderException(
```

Comment:
The decoder exception is stackless but the IAE that is the `cause` is not. OK?

Reply:
Yes, `HttpRequestMethod.of` may be invoked by users from any place they want, it might be useful to know the stack-trace there. But here we are coming from the network side, the stack-trace won't give users much info.